### PR TITLE
Use CardIO as a Cocoapod dependency

### PR DESCRIPTION
**Pull request by jmikulka**
To avoid downloading 3rd party dependency (CardIO) for the iOS project in the npm package, let's use Cocoapods to fetch the official CardIO pod for us. By doing this, we make the npm package very small and limited to the native bridge work.

### CODE REVIEW

**Comment by karanjthakkar** [comment]
I'm a little confused:
- where is the ios dependency going to be pulled from? From your cocoapods registry, or the upstream repo owners cocoapods registry? 
- is there a reason you are deleting the files from the repo? When publishing the module to npm, you can choose to skip/exclude the entire ios project from the package by adding it to `.npmignore` at the root.

**Comment by jmikulka** [comment]
1) Yes, it's going to be pulled from the public cocoapods registry: https://cocoapods.org/pods/CardIO
2) Yes, I could ignore those files, but to avoid any confusion which CardIO files are being used, I decided to remove it. That also makes the repo much smaller to clone.

File: RNAwesomeCardIO.podspec [hunk 1]
```diff
@@ -10,10 +10,10 @@ Pod::Spec.new do |s|
   s.homepage           = package["homepage"]
   s.authors            = { package["author"]["name"] => package["author"]["email"] }
   s.source             = { :git => s.homepage, :tag => "v#{s.version}" }
-  s.source_files       = ["ios/*.{h,m}", "ios/CardIO/*.h"]
+  s.source_files       = "ios/*.{h,m}"
   s.platform           = :ios, "8.0"
   s.frameworks         = "Accelerate", "AVFoundation", "AudioToolbox", "CoreMedia", "CoreVideo", "MobileCoreServices", "OpenGLES", "QuartzCore", "Security", "UIKit"
   s.libraries          = "c++"
```
Comment (a-dilettante) [inline RNAwesomeCardIO.podspec]: now that OpenCV is gone, do we need the `s.libraries` declaration?

It's listen in the CardIO Podspec: https://github.com/card-io/card.io-iOS-SDK/blob/master/CardIO.podspec#L15

**Review by a-dilettante — APPROVED** [review]
LGTM

File: RNAwesomeCardIO.podspec (hunk 1 above)
Comment (karanjthakkar) [reply RNAwesomeCardIO.podspec]: I guess this whole podspec can be deleted. It's useless now 🤔

**Review by karanjthakkar — APPROVED** [review]

File: RNAwesomeCardIO.podspec (hunk 1 above)
Comment (karanjthakkar) [reply RNAwesomeCardIO.podspec]: Oh wait I was misunderstanding. My bad.

**Review by emmajam — APPROVED** [review]
ship it

**Pull request merged**